File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    people.sort(reverse=True)
    q = deque(people)
    sumv = 0
    while q:
        if len(q) <= 1:
            q.pop()
        else:
            if q[0] + q[-1] <= limit:
                q.pop()
                q.popleft()
            else:
                q.popleft()
        answer+= 1
        
    return answer


# people.sort(reverse=True)
#     people = deque(people)
#     while people:
#         if len(people) <= 1:
#             people.pop()
#         else:
#             if people[0] + people[-1] <= limit:
#                 people.pop()
#                 people.popleft()
#             else:
#                 people.popleft()

#         answer += 1

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;
    people.sort((a, b) => a-b)
    let left = 0
    let right = people.length -1
    while(left <= right) {
        if (people[left] + people[right] <= limit) {
            left += 1
        }
        right -= 1
        answer += 1
    }
    return answer;
}